nexus/src/app/sagas/instance_create.rs

```diff
@@ -758,64 +735,52 @@ async fn ensure_instance_disk_attach_state(
     attached: bool,
 ) -> Result<(), ActionError> {
     let osagactx = sagactx.user_data();
-    let disk_params = sagactx.saga_params::<DiskParams>()?;
-    let saga_params = disk_params.saga_params;
-    let disk_index = disk_params.which;
-    let opctx =
-        OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
-
-    let saga_disks = &saga_params.create_params.disks;
-    let instance_name =
-        db::model::Name(saga_params.create_params.identity.name);
+    let params = sagactx.saga_params::<DiskAttachParams>()?;
+    let datastore = osagactx.datastore();
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
+    let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
```

Comment:
So @smklein, @andrewjstone this lookup is failing to resolve which is causing the tests to hang. It's not clear to me _why_ this isn't resolved. 

<details>
<summary>Stack trace</summary>

```
thread 'integration_tests::snapshots::test_snapshot' panicked at 'no ancestor called "instance_id"', /Users/just-be/.cargo/registry/src/github.com-1ecc6299db9ec823/steno-0.2.0/src/saga_exec.rs:1914:32
stack backtrace:
   0: rust_begin_unwind
             at /rustc/8b705839cd656d202e920efa8769cbe43a5ee269/library/std/src/panicking.rs:556:5
   1: core::panicking::panic_fmt
             at /rustc/8b705839cd656d202e920efa8769cbe43a5ee269/library/core/src/panicking.rs:142:14
   2: steno::saga_exec::ActionContext<UserType>::lookup::{{closure}}
             at /Users/just-be/.cargo/registry/src/github.com-1ecc6299db9ec823/steno-0.2.0/src/saga_exec.rs:1914:32
   3: core::option::Option<T>::unwrap_or_else
             at /rustc/8b705839cd656d202e920efa8769cbe43a5ee269/library/core/src/option.rs:825:21
   4: steno::saga_exec::ActionContext<UserType>::lookup
             at /Users/just-be/.cargo/registry/src/github.com-1ecc6299db9ec823/steno-0.2.0/src/saga_exec.rs:1911:20
   5: omicron_nexus::app::sagas::instance_create::ensure_instance_disk_attach_state::{{closure}}
             at ./src/app/sagas/instance_create.rs:741:23
```

</details>

`sic_create_instance_record` performs the same lookup. The ordering seems fine... I feel like I'm missing something basic here, I'm just not sure what.

Reply:
@smklein cleared this up for me in chat. So, for some reason in my head I was thinking that this was just a node in the top level instance graph and not a subsaga. It _is_ a subsaga though meaning this lookup wouldn't work because it's in the parent's saga, not the current saga. @smklein pointed out I could just pass `instance_id` down as a param, so I'm going to do that.